tests/specs/benchmarks/verification.k

```diff
@@ -26,6 +26,9 @@ module VERIFICATION
     imports VERIFICATION-COMMON
     imports VERIFICATION-JAVA
 
+    rule M:Map [ K1 <- _  ] [ K2 <- V2 ] => M [ K1 <- V2 ]              requires K1 ==Int K2
+    rule M:Map [ K1 <- V1 ] [ K2 <- V2 ] => M [ K2 <- V2 ] [ K1 <- V1 ] requires K2  <Int K1
+
```

Comment:
@ehildenb @denis-bogdanas I realized that this doesn't work.  Since the symbolic map is used only for the storage for the haskell backend version of KEVM, and the storage keys are the hashed location, whose ordering is unclear.  So, I think we can just keep the nested map update terms without simplifying, because usually the storage is not much updated.  Or, I can bring the simplification lemmas used for the java backend here and profile the performance overhead.  Anyway, this is the custom lemmas for the benchmarks proof, so the effect should be minimal.

Reply:
It might be worth benchmarking performance with those lemmas. If benchmarks become faster, chances are that ERC20 will too.

Reply:
Well we can also just leave them here in the benchmarks proofs for now.

Reply:
But these lemmas _will_ help with local memory, if you change it from `M:Map` to `M:Memory`. Then `_[_<-_]` is interpreted at Map-update on Java backend, and Bytes-update on Haskell backend.

Reply:
Oh, that's right. In that case, it would be better to replace this with yours:
https://github.com/kframework/evm-semantics/pull/860#discussion_r451641065

We will do that when we get there (probably other benchmarks proofs would need that.)

Reply:
I will need those lemmas for the MCD proofs, once I'm ready to merge those.